src/webots/nodes/utils/WbNodeUtilities.cpp

```diff
@@ -775,6 +779,36 @@ bool WbNodeUtilities::isDescendantOfBillboard(const WbNode *node) {
   return false;
 }
 
+bool WbNodeUtilities::isDescendantOfPropeller(const WbNode *node) {
+  if (node == NULL)
+    return false;
+
+  const WbBaseNode *initialNode = dynamic_cast<const WbBaseNode *>(node);
+
+  if (!initialNode)
+    return false;
+
+  if (initialNode->nodeType() == WB_NODE_PROPELLER)
+    return true;
+
+  WbNode *n = node->parentNode();
+  WbField *field = node->parentField(true);
+  while (n && !n->isWorldRoot() && field) {
+    WbBaseNode *baseNode = dynamic_cast<WbBaseNode *>(field->parentNode());
+
+    if (!baseNode)
+      return false;
+
+    if (baseNode->nodeType() == WB_NODE_PROPELLER)
+      return true;
+
+    field = n->parentField(true);
+    n = n->parentNode();
+  }
+
+  return false;
+}
```

Comment:
I see that this is copied from the `isDescendantOfBillboard` function.
But why do you need to also use the `WbNode::parentField()` (that is much more costly than just calling `WbNode::parentNode()`)?

Reply:
No idea, my guess is that I took example on `WbVrmlNodeUtilities::isFieldDescendant` to first implement `isDescendantOfBillboard`.
I removed the mention to fields in both `isDescendantOfBillboard` and `isDescendantOfPropeller`